File: okj/szoftverfejleszto/nobel/Nobel.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class Nobel {

    @SuppressWarnings("boxing")
    public static void main(String[] args) throws IOException {
        var lines = Files.readAllLines(Path.of("nobel.csv"));
        var dijak = new ArrayList<Dij>();
        
        for(var i = 1; i < lines.size(); ++i) {
            dijak.add(new Dij(lines.get(i)));
        }
        
        for(var dij : dijak) {
            if(dij.keresztNev.equals("Arthur B.") && dij.vezetekNev.equals("McDonald")) {
                System.out.println("3. Feladat: Arthur " + dij.tipus + " díjat kapott");
                break;
            }
        }
        
        System.out.println("4. Feladat:");
        for(var dij : dijak) {
            if(dij.ev == 2017 && dij.tipus.equals("irodalmi")) {
                System.out.println("Irodalmi díjat kapott: " + dij.keresztNev + " " + dij.vezetekNev);
            }
        }
        
        System.out.println("5. Feladat:");
        for(var dij : dijak) {
            if(dij.ev >= 1990 && dij.vezetekNev.equals("")) {
                System.out.println(dij.ev + ": " + dij.keresztNev);
            }
        }
        
        System.out.println("6. Feladat");
        for(var dij : dijak) {
            if(dij.vezetekNev.contains("Curie")) {
                System.out.println(dij.ev + ": " + dij.keresztNev + " " + dij.vezetekNev + ": " + dij.tipus);
            }
        }
        
        System.out.println("7. Feladat");
        var tipusSzamlalok = new HashMap<String, Integer>();
        for(var dij : dijak) {
            tipusSzamlalok.put(dij.tipus, tipusSzamlalok.getOrDefault(dij.tipus, 0) + 1);
        }
        
        for(var entry : tipusSzamlalok.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue() + " db");
        }
        
        dijak.sort(Comparator.comparingInt(k -> k.ev));
        
        try(var output = new PrintWriter("orvosi.txt")){
            for(var dij : dijak) {
                if(dij.tipus.equals("orvosi")) {
                    output.println(dij.ev + ":" + dij.keresztNev + " " + dij.vezetekNev);
                }
            }
        }
    }
    
    public static class Dij{
        public final int ev;
        public final String tipus;
        public final String keresztNev;
        public final String vezetekNev;
        
        public Dij(String sor) {
            var split = sor.split(";");
            
            ev = Integer.parseInt(split[0]);
            tipus = split[1];
            keresztNev = split[2];
            vezetekNev = split.length == 4 ? split[3] : "";
        }
    }
}

File: okj/szoftverfejleszto/nobel/c#/Nobel.cs
using System;
using System.IO;
using System.Collections.Generic;

var lines = File.ReadAllLines("nobel.csv");
var dijak = new List<Dij>();

for(var i = 1; i < lines.Length; ++i) {
    dijak.Add(new Dij(lines[i]));
}

foreach(var dij in dijak) {
    if(dij.keresztNev == "Arthur B." && dij.vezetekNev == "McDonald") {
        Console.WriteLine("3. Feladat: Arthur " + dij.tipus + " díjat kapott");
        break;
    }
}

Console.WriteLine("4. Feladat:");
foreach(var dij in dijak) {
    if(dij.ev == 2017 && dij.tipus == "irodalmi") {
        Console.WriteLine("Irodalmi díjat kapott: " + dij.keresztNev + " " + dij.vezetekNev);
    }
}

Console.WriteLine("5. Feladat:");
foreach(var dij in dijak) {
    if(dij.ev >= 1990 && dij.vezetekNev == "") {
        Console.WriteLine(dij.ev + ": " + dij.keresztNev);
    }
}

Console.WriteLine("6. Feladat");
foreach(var dij in dijak) {
    if(dij.vezetekNev.Contains("Curie")) {
        Console.WriteLine(dij.ev + ": " + dij.keresztNev + " " + dij.vezetekNev + ": " + dij.tipus);
    }
}

Console.WriteLine("7. Feladat");
var tipusSzamlalok = new Dictionary<string, int>();
foreach(var dij in dijak) {
    if(tipusSzamlalok.ContainsKey(dij.tipus)) {
        tipusSzamlalok[dij.tipus] = tipusSzamlalok[dij.tipus] + 1;
    }else{
        tipusSzamlalok[dij.tipus] = 0;
    }
}

foreach(var entry in tipusSzamlalok) {
    Console.WriteLine(entry.Key + ": " + entry.Value + " db");
}

dijak.Sort((k, l) => k.ev.CompareTo(l.ev));

using var output = new StreamWriter("orvosi.txt");
foreach(var dij in dijak) {
    if(dij.tipus == "orvosi") {
        output.WriteLine(dij.ev + ":" + dij.keresztNev + " " + dij.vezetekNev);
    }
}